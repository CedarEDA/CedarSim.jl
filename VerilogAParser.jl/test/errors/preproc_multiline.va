`define MPRco(nam,def,uni,lwr,upr,des) (*units=uni, desc=des*) parameter real nam=def from[lwr:upr) ;
module preproc_multiline(p, n);
    `MPRco( VBX           , PHI - 7.7348e-4*NDEP * XT * XT
                                           ,"V"        ,0           ,inf,        ,"Vbs at which the depletion region width equals XT")
endmodule