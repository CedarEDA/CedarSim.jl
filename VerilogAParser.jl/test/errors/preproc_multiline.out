[31m[1m╭ ERROR [VAS01]:[22m[39m[31m Macro expansion argument mismatch.[39m
[31m│[39m[90m  ╭───── preproc_multiline.va:3[39m
[31m│[90m3[0m [90m│[0m     `MPRco( VBX           , PHI - 7.7348e-4*NDEP * XT * XT
[31m│[0m  [90m┊[0m            [92m━━━━━━━┳━━━━━━━[0m [36m━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━┅
[31m│[0m  [90m┊[0m                  [92mnam[0m                     [36mdef[0m
[31m│[90m4[0m [90m│[0m                                            ,"V"        ,0           ,inf,        ,[91m"Vbs at which the depletion region width equals XT"[0m)
[31m│[0m  [90m┊[0m [36m┅━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━[0m [93m━━━━━┳━━━━━[0m [35m━━━━━━┳━━━━━[0m [94m━┳━[0m [95m━━━━┳━━━[0m [91m━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━
[31m│[0m  [90m┊[0m                     [36mdef[0m                         [93muni[0m          [35mlwr[0m     [94mupr[0m    [95mdes[0m                    [91m✗ Extra Argument[0m
[31m│[39m[90m  │[39m
[31m│[39m[90m  ╰───── [39mMacro [39m[1mMPRco[22m was called with 7 arguments.
[31m│[39m
[31m│[39m[90m  ╭───── preproc_multiline.va:1[39m
[31m│[90m1[0m [90m│[0m `define MPRco([92mnam[0m,[36mdef[0m,[93muni[0m,[35mlwr[0m,[94mupr[0m,[95mdes[0m) (*units=[93muni[0m, desc=[95mdes[0m*) parameter real [92mnam[0m=[36mdef[0m from[[35mlwr[0m:[94mupr[0m) ;
[31m│[39m[90m  │[39m
[31m│[39m[90m  ╰───── [39mMacro [39m[1mMPRco[22m takes 6 arguments.
[31m│[39m
[31m│[39m[34m help: [39mThe macro [39m[1mMPRco[22m was called with more arguments (7) than it expected (6).
[31m│[39m       Check for any misplaced commas in the macro invocation above.
[31m╯[39m
